sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/DeploymentStep.java

```diff
@@ -181,17 +181,17 @@ private Status getStatus(Map<Protos.TaskID, TaskStatusPair> tasks) {
             return Status.PENDING;
         }
 
+        Set<Status> statuses = new HashSet<>();
         for (Map.Entry<Protos.TaskID, TaskStatusPair> entry : tasks.entrySet()) {
             String taskId = entry.getKey().getValue();
             Status status = entry.getValue().getStatus();
             logger.info("TaskId: {} has status: {}", taskId, status);
-            if (!status.equals(Status.COMPLETE)) {
-                // Keep and log current status
-                return super.getStatus();
-            }
+
+            statuses.add(status);
         }
 
-        return Status.COMPLETE;
+
+        return Collections.min(statuses);
```

Comment:
This implies that the statuses have a concept of ordering -- I assume that this is defined by the order of declaration of the enum class. Is there a particular ordering that we depend on for this to work correctly here? For example, if one task is `COMPLETE` and another is `IN_PROGRESS` this here will return `COMPLETE`? It would be good to get some unit tests in place that capture the desired behaviour for the `Enum` then.

Reply:
DeploymentStep will only ever return ERROR through COMPLETE. IN_PROGRESS is reserved for steps and phases.

Yes, there should be a unit test for this. It _is_ exercised by different unit tests, but I will make the `getStatus` method package-private so I can hit it in a unit test and validate it fully..